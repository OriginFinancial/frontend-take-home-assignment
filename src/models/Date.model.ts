export interface MonthYear {
  month: string;
  year: number;
}
