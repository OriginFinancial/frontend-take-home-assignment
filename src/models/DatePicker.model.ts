import { KeyboardEvent, MonthYear } from '.';

export interface DatePickerProps {
  data: Data;
  actions: Actions;
}

interface Data {
  selectedDate: MonthYear;
  isPrevButtonDisabled: boolean;
}

interface Actions {
  handlePressNextDate: () => void;
  handlePressPreviousDate: () => void;
  handlePressKeyboardArrows: (e: KeyboardEvent) => void;
}
