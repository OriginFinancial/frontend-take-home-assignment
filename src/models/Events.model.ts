export interface KeyboardEvent {
  keyCode: number;
}

export interface FormEvent {
  e: React.FormEvent<HTMLInputElement>;
}
