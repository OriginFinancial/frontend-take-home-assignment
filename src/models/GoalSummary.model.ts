import { MonthYear } from '.';
export interface GoalSummaryProps {
  data: Data;
}

interface Data {
  amount: string;
  selectedDate: MonthYear;
  totalOfMonths: number;
  monthlyAmount: string;
}
