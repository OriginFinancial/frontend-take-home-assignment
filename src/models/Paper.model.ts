export interface PaperProps {
  children: ChildrenProps;
  radiusSmall?: boolean;
}

export interface ChildrenProps {
  children?: JSX.Element[] | JSX.Element | React.ReactNode;
}
