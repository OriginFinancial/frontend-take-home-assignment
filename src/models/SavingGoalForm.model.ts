import { KeyboardEvent, MonthYear } from '.';

export interface SavingGoalFormProps {
  data: Data;
  actions: Actions;
}

interface Data {
  amount: string;
  selectedDate: MonthYear;
  isPrevButtonDisabled: boolean;
  totalOfMonths: number;
  monthlyAmount: string;
}

interface Actions {
  handleInputChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  handlePressNextDate: () => void;
  handlePressPreviousDate: () => void;
  handlePressKeyboardArrows: (e: KeyboardEvent) => void;
}
