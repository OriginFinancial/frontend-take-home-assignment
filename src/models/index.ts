export * from './Date.model';
export * from './DatePicker.model';
export * from './Events.model';
export * from './GoalSummary.model';
export * from './SavingGoalForm.model';
export * from './Paper.model';
export * from './TextInput.model';
