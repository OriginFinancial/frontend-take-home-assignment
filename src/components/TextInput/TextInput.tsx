import * as React from 'react';
import styled from 'styled-components';
import { formComponentFocus } from '../../styles';
import { TextInputProps } from '../../models';

const TextInput: React.FunctionComponent<TextInputProps> = ({
  label,
  value,
  handleChange
}) => {
  return (
    <Wrapper>
      <Label htmlFor="numberInput">{label}</Label>

      <Group>
        <DollarSign>$</DollarSign>

        <Input
          value={value}
          onChange={handleChange}
          autoFocus
          type="text"
          id="numberInput"
        />
      </Group>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  width: 100%;
`;

const Label = styled.label`
  display: inline-block;
  font-weight: 700;
  margin-bottom: 0.8rem;
`;

const Group = styled.div`
  align-items: center;
  display: flex;
  position: relative;
`;

const Input = styled.input`
  background-color: ${props => props?.theme?.colors?.white};
  border: 0.1rem solid ${props => props?.theme?.colors?.primary?.extralight};
  border-radius: 0.3rem;
  color: ${props => props?.theme?.colors?.text?.main};
  font-weight: 700;
  height: 5rem;
  padding-left: 7rem;
  width: 100%;

  ${formComponentFocus};
`;

const DollarSign = styled.div`
  align-items: center;
  border: 0.1rem solid ${props => props?.theme?.colors?.secondary?.main};
  border-bottom-left-radius: 0.4rem;
  border-top-left-radius: 0.4rem;
  background-color: ${props => props?.theme?.colors?.primary?.extralight};
  color: ${props => props?.theme?.colors?.text?.light};
  display: flex;
  font-size: 1.8rem;
  height: 4.8rem;
  justify-content: center;
  left: 0.1rem;
  position: absolute;
  width: 6rem;
`;

export default TextInput;
