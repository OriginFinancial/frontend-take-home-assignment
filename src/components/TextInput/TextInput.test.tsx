import * as React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import { TextInputProps } from '../../models';
import TextInput from './TextInput';

describe('TextInput component', () => {
  let component: ReactWrapper;
  let props: TextInputProps;

  beforeEach(() => {
    jest.clearAllMocks();
    const handleChange = jest.fn();
    props = {
      value: '1,000',
      label: 'Total amount',
      handleChange
    };
    component = mount(<TextInput {...props} />);
  });

  it('renders initial layout', () => {
    expect(component.getElements()).toMatchSnapshot();
  });

  it('renders the label prop', () => {
    expect(component.find('label').text()).toBe(props.label);
  });

  it('changes the value when handleChange is called', () => {
    expect(
      component
        .find('input#numberInput')
        .simulate('change', {
          currentTarget: { value: '1,000' }
        })
        .props().value
    ).toBe('1,000');
  });
});
