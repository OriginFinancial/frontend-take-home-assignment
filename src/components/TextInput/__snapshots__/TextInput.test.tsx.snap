// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput component renders initial layout 1`] = `
Array [
  <TextInput
    handleChange={[MockFunction]}
    label="Total amount"
    value="1,000"
  />,
]
`;
